ext/net/ops_tls.rs

```diff
@@ -1051,11 +1051,21 @@ where
     return Err(generic_error("`key` is not specified."));
   };
 
-  let mut tls_config = ServerConfig::builder()
+  let mut tls_config_result = ServerConfig::builder()
     .with_safe_defaults()
     .with_no_client_auth()
-    .with_single_cert(cert_chain, key_der)
-    .expect("invalid key or certificate");
```

Comment:
Instead of using a match, you can use `map_err` and the `?` operator to return early:

`.with_single_cert(...).map_err(|e| ...)?`

Reply:
@mmastrac Thank you for the suggestion! I've [refactored ](https://github.com/denoland/deno/pull/20157/commits/6c46040283844b37a96cce2bc7d632b878f9bd16) the code to use `map_err` instead of a match expression. Are these changes suitable?

Reply:
Some changes requested inline. We'll need to add two tests to `tls_test.ts` and ensure they pass before we can merge this, however. 

```
Deno.test(
  { permissions: { net: true, read: false } },
  function listenTLSInvalidCert() {
    assertThrows(() => {
      Deno.listenTls({
        hostname: "localhost",
        port: 3500,
        certFile: "cli/tests/testdata/tls/invalid.crt",
        keyFile: "cli/tests/testdata/tls/localhost.key",
      });
    }, Deno.errors.InvalidData);
  },
);

Deno.test(
  { permissions: { net: true, read: false } },
  function listenTLSInvalidKey() {
    assertThrows(() => {
      Deno.listenTls({
        hostname: "localhost",
        port: 3500,
        certFile: "cli/tests/testdata/tls/localhost.crt",
        keyFile: "cli/tests/testdata/tls/invalid.key",
      });
    }, Deno.errors.InvalidData);
  },
);

```

Reply:
Added the two tests with [this commit](https://github.com/denoland/deno/pull/20157/commits/ff16be0772a5a7356696302a9efe3c80e01117a5)

Reply:
Great! You'll also need to add an invalid key/crt file to match those files we specified here (otherwise the error generated will be incorrectly "NotFound")

Thanks for working on this!

Reply:
I've added an invalid key and an invalid crt file with [this commit](https://github.com/denoland/deno/pull/20157/commits/0591653947d669fefdb6107c75b1765d5e462dd3)

Thank you for all your help @mmastrac! I really appreciate it as this is my first "official" open source contribution and I couldn't be happier that it is with Deno :smile:

Reply:
Awesome. :) I approved the build run and fingers crossed it works as we expect. In the meantime, you can run `cargo test` locally and you should see your new tests pass.